sql/src/main/java/org/apache/druid/sql/calcite/schema/DruidSchema.java

```diff
@@ -494,14 +633,17 @@ void removeServerSegment(final DruidServerMetadata server, final DataSegment seg
 
   private long recomputeIsRealtime(ImmutableSet<DruidServerMetadata> servers)
   {
+    if (servers.isEmpty()) {
+      return 0;
+    }
     final Optional<DruidServerMetadata> historicalServer = servers
         .stream()
-        .filter(metadata -> metadata.getType().equals(ServerType.HISTORICAL))
+        .filter(metadata -> metadata.getType().equals(ServerType.HISTORICAL)
+                            || metadata.getType().equals(ServerType.BROKER))
```

Comment:
Yeah, I knew the current behavior but thought this could be better. But, on the second thought, probably better to keep the code logic consistent, so I reverted this change and added some comment.